source/client.c

```diff
@@ -238,6 +258,8 @@ static void s_attempt_reconect(struct aws_task *task, void *userdata, enum aws_t
     if (status == AWS_TASK_STATUS_RUN_READY && connection) {
         /* If the task is not cancelled and a connection has not succeeded, attempt reconnect */
 
+        AWS_LOGF_TRACE(AWS_LS_MQTT_CLIENT, "Attempting reconnect");
```

Comment:
It may be worth logging the no-reconnect case and why (max exceeded, canceled, etc...)

Reply:
There isn't a no-reconnect case